sklearnex/neighbors/_lof.py

```diff
@@ -0,0 +1,180 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import warnings
+
+import numpy as np
+from sklearn.neighbors import LocalOutlierFactor as sklearn_LocalOutlierFactor
+from sklearn.utils.validation import check_is_fitted
+
+from daal4py.sklearn._utils import control_n_jobs, run_with_n_jobs, sklearn_check_version
+from onedal.neighbors import NearestNeighbors as onedal_NearestNeighbors
+
+from .._device_offload import dispatch, wrap_output_data
+from .common import KNeighborsDispatchingBase
+
+
+@control_n_jobs
+class LocalOutlierFactor(KNeighborsDispatchingBase, sklearn_LocalOutlierFactor):
+    __doc__ = sklearn_LocalOutlierFactor.__doc__
+    if sklearn_check_version("1.2"):
+        _parameter_constraints: dict = {
+            **sklearn_LocalOutlierFactor._parameter_constraints
+        }
+
+    def __init__(
+        self,
+        n_neighbors=20,
+        *,
+        algorithm="auto",
+        leaf_size=30,
+        metric="minkowski",
+        p=2,
+        metric_params=None,
+        contamination="auto",
+        novelty=False,
+        n_jobs=None,
+    ):
+        super().__init__(
+            n_neighbors=n_neighbors,
+            algorithm=algorithm,
+            leaf_size=leaf_size,
+            metric=metric,
+            p=p,
+            metric_params=metric_params,
+            n_jobs=n_jobs,
+            contamination=contamination,
+            novelty=novelty,
+        )
+
+    @run_with_n_jobs
+    def _onedal_fit(self, X, y, queue=None):
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        onedal_params = {
+            "n_neighbors": self.n_neighbors,
+            "algorithm": self.algorithm,
+            "metric": self.effective_metric_,
+            "p": self.effective_metric_params_["p"],
+        }
+
+        try:
+            requires_y = self._get_tags()["requires_y"]
+        except KeyError:
+            requires_y = False
+
+        self._onedal_estimator = onedal_NearestNeighbors(**onedal_params)
+        self._onedal_estimator.requires_y = requires_y
+        self._onedal_estimator.effective_metric_ = self.effective_metric_
+        self._onedal_estimator.effective_metric_params_ = self.effective_metric_params_
+        self._onedal_estimator.fit(X, y, queue=queue)

```

Comment:
Should we make it a separate function? It looks like a separate sequence of actions

Reply:
You caught me, its a copy-paste of the NearestNeighbors._onedal_fit() function into this